10.9-libcxx/stable/main/finkinfo/editors/emacsen-common.info

```diff
@@ -1,42 +1,84 @@
 SourceDirectory: %n-%v
-ConfFiles: %p/etc/emacs/site-start.el
+# ConfFiles: %p/etc/emacs/site-start.el
 PatchFile: %n.patch
-PatchFile-Checksum: SHA256(1533417a3512c1f61b7c0c2ee26014e51492059f7602a26c4fe8e0f4b116575b)
+PatchFile-Checksum: SHA256(1bc2385910e96102cb18976b0435df71285f2b892e6d2f50115d3fff8c11fad8)
 PatchScript: sed 's|@PREFIX@|%p|g' <%{PatchFile} | patch -p1
 DocFiles: debian/changelog debian-emacs-policy debian/copyright
 CompileScript: <<
   perl -wc emacs-install
+  perl -wc emacs-remove
   perl -wc emacs-package-install
-  perl -wc generate-install-list
+  perl -wc emacs-package-remove
+  perl -wc lib.pl
 <<
 InstallScript: <<
  mkdir -p %i/etc/emacs/site-start.d
  mkdir -p %i/etc/site-start.d
  mkdir -p %i/lib/%n/packages/install
  mkdir -p %i/lib/%n/packages/remove
+ mkdir -p %i/lib/%n/packages/compat
+ mkdir -p %i/var/lib/%n/state/flavor/installed
+ mkdir -p %i/var/lib/%n/state/package/installed
  mkdir -p %i/share/doc/%n
  mkdir -p %i/share/emacs/site-lisp
 
- install -m 644 site-start.el %i/etc/emacs/
- install -m 644 00debian-vars.el %i/etc/emacs/site-start.d/00fink-vars.el
  install -m 755 emacs-install %i/lib/%n/
+ install -m 755 emacs-remove %i/lib/%n/
  install -m 755 emacs-package-install %i/lib/%n/
- (cd %i/lib/%n/; ln -s emacs-install emacs-remove)
- (cd %i/lib/%n/; ln -s emacs-package-install emacs-package-remove)
- install -m 755 generate-install-list %i/lib/%n/
+ install -m 755 emacs-package-remove %i/lib/%n/
+ install -m 755 lib.pl %i/lib/%n/
  install -m 644 debian-startup.el %i/share/emacs/site-lisp/fink-startup.el
+ install -m 644 compat/emacsen-common %i/lib/%n/packages/compat/
 
  install -m 755 emacsen-common.install %i/lib/%n/packages/install/emacsen-common
  install -m 755 emacsen-common.remove %i/lib/%n/packages/remove/emacsen-common 
 <<
+PreInstScript: <<
+# When emacsen-common and add-ons are upgrading simultaneously,
+# emacsen-common can be unpacked but not configured when an add-on's
+# prerm runs. That causes the add-on's removal command to be skipped,
+# which is wrong and can cause the upgrade to fail. For example
+# https://bugs.debian.org/1106291
 #
+# For now, to ensure the removal commands are still run, remove all
+# emacs flavors here, keeping track of which flavors are actually
+# still installed in pending, and then restore them from the postinst,
+# when emacsen-common is ready again.
+#
+# Normally, it would be unsafe to depend on emacsen-common files like
+# this in the preinst, but our assumption for this mitigation is that
+# since emacs flavors depend on emacsen-common and since they should
+# have a installed/FLAVOR file if and only if they're "ready to go"
+# (i.e. via their postinst configure), then the emacs-remove script
+# will be available inside the loop.
+
+mkdir -p %p/var/lib/emacsen-common/state/flavor/pending
+for flavor in %p/var/lib/emacsen-common/state/flavor/installed/*[!~]; do
```

Comment:
For me on 13.7/x86_64 with only emac26 installed, the installed versions are in the physical file `%p/var/lib/emacsen-common/installed-flavors`. The above `.../state/flavor/installed/...` path doesn't exist. Is this path something that started with emacs28 and/or emacsen-common-3 ?